src/track/macroaction.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include "engine/engine.h"
+#include "proto/macro.pb.h"
+namespace proto = mixxx::track::io;
+
+/// A MacroAction is the building block of a Macro.
+/// It contains a position as well as the action to be taken at that position.
+///
+/// Note that currently only jumps to a target position are available, but that
+/// is subject to change.
+class MacroAction {
+  public:
+    enum class Type : uint32_t {
+        Jump = 0
+    };
+
+    MacroAction(double sourceFramePos, double targetFramePos)
+            : sourceFrame(sourceFramePos), targetFrame(targetFramePos), type(Type::Jump){};
+
+    Type getType() const {
+        return type;
+    }
+
+    double getSourcePosition() const {
```

Comment:
getSourcePositionFrames?

Reply:
Frames should be the default measurement for positions, we discussed this on Zulip at some point. We should forget Samples for everything outside raw audio interaction.

Reply:
I was referring to renaming the method to make it clear what unit is used.

Reply:
But we should have a default unit that needs no specification, or not?